File: mayormenor.php
<?php
    $num1 = readline ("Digite el primer numero: ");
    $num2 = readline ("Digite el segundo numero: ");

    if($num1 > $num2){
        echo "El mayor es: $num1 \n";
        echo "La diferencia entre el numero 1 y el numero 2 es: " . ($num1 - $num2);
    }else if($num1 == $num2){
        echo "Los numeros son iguales";
    }else{
        echo "El mayor es: $num2 \n";
        echo "La diferencia entre el numero 1 y el numero 2 es: " . ($num2 - $num1);
    }
?>

File: MayorMenor.java
import java.util.Scanner;

public class MayorMenor{
    public static void main(String[] args){
        Scanner keyboard = new Scanner(System.in);

        System.out.println("Ingrese el primer numero");
        int num1 = keyboard.nextInt();

        System.out.println("Ingrese el segundo numero");
        int num2 = keyboard.nextInt();

        keyboard.close();

        if(num1 > num2){
            System.out.println("El numero mayor es: " + num1 + " y la diferencia entre ambos es: " + (num1 - num2));
        }else if(num1 == num2){
            System.out.println("Los numeros son iguales y la diferencia entre ambos es: 0");
        }else{
            System.out.println("El numero mayor es: " + num2 + " y la diferencia entre ambos es: " + (num2 - num1));
        }
    }
}